design/protected-apps-view.md

```diff
@@ -0,0 +1,917 @@
+
+Rather than having correlation logic scattered across multiple frontend hooks, we create a single authoritative source of truth: a new Custom Resource Definition called ProtectedApplicationView. This CRD represents the aggregated information about a protected application, regardless of its type. Every protected application gets one corresponding ProtectedApplicationView resource, and the frontend simply watches and displays these views.
+
+This approach provides strong consistency guarantees. When a user sees a ProtectedApplicationView resource, they know it reflects the current state of the underlying application and its DR configuration. There are no race conditions between multiple hooks trying to correlate the same data independently.
+
+
+### 4.2 Architecture Overview
+
+The proposed architecture introduces three new controllers and one new CRD. Let's examine each component and how they work together.
+
+#### 4.2.1 The ProtectedApplicationView CRD
+
+This new custom resource represents the aggregated view of a protected application. Its schema is designed to accommodate all three application types while making the common patterns easy to work with:
+
+```yaml
+apiVersion: ramendr.openshift.io/v1alpha1
+kind: ProtectedApplicationView
+metadata:
+  name: appset-payment-service  # Prefixed with type for uniqueness
+  namespace: openshift-gitops    # Same namespace as the source application
+  
+  # Owner reference ensures automatic cleanup
+  ownerReferences:
+    - apiVersion: argoproj.io/v1alpha1
+      kind: ApplicationSet
+      name: payment-service
+      uid: <applicationset-uid>
+      controller: true
+      blockOwnerDeletion: true
+  
+  # Labels enable efficient querying
+  labels:
+    app.kubernetes.io/type: ApplicationSet
+    ramendr.openshift.io/dr-policy: metro-dr-policy
+    ramendr.openshift.io/protected: "true"
+
+spec:
+  # The type determines which fields are populated
+  type: ApplicationSet  # or "Subscription" or "Discovered"
+  
+  # Reference back to the actual application resource
+  applicationRef:
+    apiVersion: argoproj.io/v1alpha1
+    kind: ApplicationSet
+    name: payment-service
+    namespace: openshift-gitops
+    uid: <applicationset-uid>
+  
+  # Placement information (common to all types)
+  # An application can have multiple placements (common with Subscriptions)
+  placements:
+    - name: payment-service-placement
+      namespace: openshift-gitops
+      kind: Placement
+      
+      # The clusters where this placement has scheduled the application
+      selectedClusters:
+        - primary-datacenter
+      
+      numberOfClusters: 1
+  
+  # Disaster recovery configuration (if protected)
+  # This is the most important section for the UI
+  drInfo:
+    enabled: true
+    
+    # Reference to the DRPC providing protection
+    drpcRef:
+      name: payment-service-drpc
+      namespace: openshift-gitops
```

Comment:
In case of Subscription apps, how will be differentiate between what goes in `drpcRef` spec and what goes in `additionalDRPCs` status ?? can we make `drpcRef` a list itself ??

Reply:
just confirming, if there's a reason for having drpcRef and additionalDRPCs as separate fields...

Reply:
can user perform actions (like failover/relocate/disable-dr etc) on partial subs app ??
eg: in above scenario, they just failover/relocate/disable-dr on `sub3` (drpc2) and not `sub1 and sub2` (drpc1)

Is that valid or is that not allowed ??